Adding custom testing based on opinfos input for ops with custom rules.

Gamrix opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #67575
* __->__ #67500
* #66973
* #66948
* #66898

Differential Revision: [D32497547](https://our.internmc.facebook.com/intern/diff/D32497547)

Gamrix committed:
Adding custom testing based on opinfos input for ops with custom rules.

[ghstack-poisoned]

Gamrix committed:
Update on "Adding custom testing based on opinfos input for ops with custom rules."




[ghstack-poisoned]

Gamrix committed:
Update on "Adding custom testing based on opinfos input for ops with custom rules."




[ghstack-poisoned]

Krovatkin commented on test/jit/test_dtype_analysis.py:
```diff
@@ -164,10 +177,59 @@ def test_binary_scalar(self):
+        weight, bias = conv_in.args
+        weight = weight.type(torch.long)
+
+        with self.assertRaises(RuntimeError):
```
can have a test to assert that dtype_propagation doesn't dtype the output tensor in this case?

Gamrix replied:
Makes sense. I'll have to add a bit more logic to handle that.

Krovatkin commented on test/jit/test_dtype_analysis.py:
```diff
@@ -164,10 +177,59 @@ def test_binary_scalar(self):
+            (conv2d_fn, sample_inputs_conv2d),
+            (adaptive_avg_pool2d_fn, sample_inputs_adaptive_avg_pool2d),
+        ):
+            for dtype in (torch.int8, torch.float64):
```
how did you decide on these two types?

Gamrix replied:
Just seemed like two types that would be commonly used, and I wanted one int, and one float type

Krovatkin reviewed (approved):
:shipit:

Gamrix committed:
Update on "Adding custom testing based on opinfos input for ops with custom rules."




[ghstack-poisoned]

Gamrix committed:
Update on "Adding custom testing based on opinfos input for ops with custom rules."




[ghstack-poisoned]

Gamrix committed:
Update on "Adding custom testing based on opinfos input for ops with custom rules."




[ghstack-poisoned]

Gamrix committed:
Update on "Adding custom testing based on opinfos input for ops with custom rules."




[ghstack-poisoned]

Gamrix commented:
@gamrix has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32175957).

Gamrix committed:
Update on "Adding custom testing based on opinfos input for ops with custom rules."


Differential Revision: [D32175957](https://our.internmc.facebook.com/intern/diff/D32175957)

[ghstack-poisoned]

Gamrix commented:
@gamrix has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32175957).

Gamrix committed:
Update on "Adding custom testing based on opinfos input for ops with custom rules."


Differential Revision: [D32175957](https://our.internmc.facebook.com/intern/diff/D32175957)

[ghstack-poisoned]

Gamrix commented:
@gamrix has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32175957).

Gamrix committed:
Update on "Adding custom testing based on opinfos input for ops with custom rules."


Differential Revision: [D32175957](https://our.internmc.facebook.com/intern/diff/D32175957)

[ghstack-poisoned]

Gamrix committed:
Update on "Adding custom testing based on opinfos input for ops with custom rules."

* #66898

[ghstack-poisoned]

Gamrix commented:
@gamrix has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32497547).
